modules/electrophysiology_browser/php/sessions.class.inc

```diff
@@ -448,6 +481,591 @@ class Sessions extends \NDB_Page
+                        AND FileType='json'",
+                        $params
+                    );
+
+                    $instance['AnnotationFileID'] = $tsv_ID;
+                    $metadata['AnnotationFileID'] = $json_ID;
+
+                    /* If no instance ID is specified, insert new instance
+                     * into instance table and get the parameter file ID
+                     * from the parameter table
+                     */
+                    if (empty($values['instance_id'])) {
+                        $parameterID = $db->pselectone(
+                            "SELECT AnnotationParameterID
+                            FROM physiological_annotation_parameter
+                            WHERE AnnotationFileID=:annotationFID",
+                            ['annotationFID' => $json_ID]
+                        );
+                        $instance['AnnotationParameterID'] = $parameterID;
+
+                        $db->insert('physiological_annotation_instance', $instance);
+                    } else {
+                        //Update instance table
+                        if (!empty($values['instance_id'])) {
+                            $db->update(
+                                'physiological_annotation_instance',
+                                $instance,
+                                ['AnnotationInstanceID' => $values['instance_id']]
+                            );
+                        }
+                        //Update parameter table
+                        if (!empty($values['parameter_id'])) {
+                            $db->update(
+                                'physiological_annotation_parameter',
+                                $metadata,
+                                ['AnnotationParameterID' => $values['parameter_id']]
+                            );
+                        }
+
+                    }
+
+                    //In all cases where files are not new,
+                    //set LastUpdate time for all related files
+                    $db->update(
+                        'physiological_annotation_file',
+                        ['LastUpdate' => date("Y-m-d H:i:s")],
+                        ['PhysiologicalFileID' => $physioFileID]
+                    );
+
+                }
+            }
+        }
+    }
+
+    /**
+     * Deletes one annotation from a given physiological file
+     *
+     * @param ServerRequestInterface $request DELETE request from server
+     *
+     * @return void
+     */
+    function _deleteAnnotation(ServerRequestInterface $request)
+    {
+        $values = $request->getQueryParams();
+        $user   = \NDB_Factory::singleton()->user();
+        $db     = \NDB_Factory::singleton()->database();
+        $params = [
+            'AnnotationFileID'     => $values['file_id'],
+            'AnnotationInstanceID' => $values['instance_id']
+        ];
+
+        if ($this->_hasEditDerivPerms($user)) {
```

Comment:
same comment as above, you don't need do do a lot of this if your user doesn't have the proper permission. do your permission check first